kytos/core/logs.py

```diff
@@ -115,6 +117,58 @@ def filter_session_disconnected(record):
                     and record.args and isinstance(record.args[0], str)
                     and record.args[0].endswith(msg_end))
 
+    @staticmethod
+    def decorate_logger_class(decorator):
+        old_class = logging.getLoggerClass()
+        new_class = decorator(old_class)
+        logging.setLoggerClass(new_class)
+
+
+def logger_queue_decorator(klass):
+    class QueueLogger(klass):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            self.queue = Queue()
+            self.queueListener = handlers.QueueListener(self.queue, respect_handler_level=True)
+            self.queueListener.start()
+            super().addHandler(handlers.QueueHandler(self.queue))
+
+        def changeHandlers(self, *hdlrs):
+            self.queueListener.stop()
+            self.queueListener = handlers.QueueListener(self.queue, *hdlrs, respect_handler_level=True)
+            self.queueListener.start()
+
+        def addHandler(self, hdlr):
+            old_handlers = self.queueListener.handlers
+            self.changeHandlers(*old_handlers, hdlr)
+
+        def removeHandler(self, hdlr) -> None:
+            old_handlers = self.queueListener.handlers
+            new_handlers = [handler for handler in old_handlers if handler is not hdlr]
+            self.changeHandlers(*new_handlers)
+
+        def hasHandlers(self) -> bool:
+            if self.queueListener.handlers:
+                return True
+            if self.propagate:
+                return self.parent.hasHandlers()
+            return False
+
+    return QueueLogger
+
+def logger_apm_decorator(klass):
```

Comment:
Rather than doing two separate options like you suggested, I implemented a new option that sort of combines both features. I added in a new option called logger_decorators that takes a list of decorator names, and resolves each of those names to the actual object. For example `kytos.core.logger_decorators.queue_decorator` resolves to the `queue_decorator` in the `kytos.core.logger_decorators` module. The resolution process is based on the `_resolve` function used by `logging.config` of python.